src/redux/data.ts

```diff
@@ -1405,7 +1444,7 @@ export const dataAddNewTransaction = (
 ) =>
   withRunExclusive(
     async (): Promise<RainbowTransaction | undefined> => {
-      const { transactions } = getState().data;
+      const { pendingTransactions } = getState().data;
```

Comment:
because we're splitting pending and confirmed I think we can remove the mutex stuff (withRunExclusive) [on line 1445](https://github.com/rainbow-me/rainbow/pull/3164/files#diff-3568b4eaf1ead5c20262106d9a20b6660576efe9097055edf9214b647742ccedR1447)

Reply:
I tested some transactions and after removing all instances of this function and things seem to be 👍.